bridge/data/:w

```diff
@@ -0,0 +1,203 @@
+import types
+
+def emptyProtocolInBuffer():
+    return types.SimpleNamespace(data = [], flags = -1, device_id = -1, data_length = 0)
+
+class ProtocolValsIterator:
+    """ Useful for iterating over the values inside this buffer """
+    def __init__(self, buff):
+        self._buffer = buff
+        self._index = 0
+
+    def __next__(self):
+        if self._index < self._buffer.data_length:
+            self._index += 1
+            return self._buffer.data[self._index]
+        else:
+            raise StopIteration
+
+
+class ProtocolBuffer:
+
+    def __init__(self):
+        self.inBuffer = emptyProtocolInBuffer()
+
+    def readChar(self, val):
+        """
+        Meant to be used in a loop to read a sequence of bytes,
+        needs to be implemented better!
+        """
+
+        if (val == b'\xfe'): #EOLi
+            return True
+
+        elif (val == b'\xff'):
+            self.inBuffer = emptyProtocolInBuffer()
```

Comment:
why does a protocol buffer has an object called inBuffer? why are those not properties of the protocol buffer itself?

Reply:
that :w should not exist and i tried to delete it but i keep finding it back in my folder :|

Reply:
this file does not exist anymore ^^